File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        self.result = []

        def inorder(node):
            if node is not None:
                inorder(node.left)
                self.result.append(node.val)
                inorder(node.right)

        inorder(root)
        if len(self.result) != len(set(self.result)):
            return False
        else:
            return self.result == sorted(self.result)

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> result = new ArrayList<>();

    public int kthSmallest(TreeNode root, int k) {
        postorder(root);
        return result.get(k - 1);
    }

    void postorder(TreeNode node) {
        if (node == null) {
            return;
        }

        postorder(node.left);
        result.add(node.val);
        postorder(node.right);
    }
}